src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1862: Prefer the StringComparison method overloads to perform case-insensitive string comparisons.
+    /// </summary>
+    public class RecommendCaseInsensitiveStringComparisonFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(RecommendCaseInsensitiveStringComparisonAnalyzer.RuleId);
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken ct = context.CancellationToken;
+            if (ct.IsCancellationRequested)
+            {
+                return;
+            }
+
+            Document doc = context.Document;
+
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode node)
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (model.GetOperation(node, ct) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison)
+                is not INamedTypeSymbol stringComparisonType)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparer)
+                is not INamedTypeSymbol stringComparerType)
+            {
+                return;
+            }
+
+            // Only diagnose overloads that take one string argument:
+            // Contains(string), IndexOf(string), StartsWith(string), CompareTo(string)
+            if (invocation.Arguments.Count() != 1)
```

Comment:
Isn't there a Length property instead of LINQ's Count method?

Reply:
I can change it.

I wonder if the perf is noticeably better in Length over Count, so that we could consider suggesting an analyzer+fixer for that.

I thought we already had one, but the closest one I found was [CA1836: "Prefer IsEmpty over Count or Length"](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1836).

Reply:
@carlossanlop That would be https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1829.

Reply:
Ok interesting. `invocation.Arguments.Count() != 1` did not give me the suggestion to use `Length`. The analyzer does not seem to diagnose `ImmutableArray`. I changed it anyway, but I wonder if the behavior is expected or not.

@dotnet/area-system-collections Hey folks, do you know the reason why?

Reply:
I opened an issue to track this: https://github.com/dotnet/roslyn-analyzers/issues/6667